### ISSUE TITLE
apply_ufunc(dask='parallelized') to infer output dtypes

### ISSUE THREAD

**Issue by crusaderky** [pydata/xarray#1698]
If one doesn't provide the ``dtype`` parameter to ``dask.map_blocks()``, it automatically infers it by running the kernel on trivial dummy data.
It should be straightforward to make ``xarray.apply_ufunc(dask='parallelized')`` use the same functionality if the ``output_dtypes`` parameter is omitted.

**Comment by shoyer** [comment]
Yes, this would be a reasonable this to do, although we should consider adding it upstream in `dask.array.atop` instead. `atop` is what powers `dask='parallelized'` under the hood.

**Comment by crusaderky** [comment]
Still valid
